test/integration/chain_of_trust.js

```diff
@@ -61,13 +64,25 @@ suite('certificate of trust', () => {
     assert.equal(result.run.reasonResolved, 'completed', 'task should be successful');
 
     let expectedArtifacts = ['public/logs/certified.log',
+      'public/chain-of-trust.json',
+      'public/chain-of-trust.json.sig',
       'public/chainOfTrust.json.asc',
       'public/logs/live.log',
       'public/logs/live_backing.log',
       'public/xfoo',
       'public/bar'].sort();
     assert.deepEqual(Object.keys(result.artifacts).sort(), expectedArtifacts);
 
+    // ed25519 cot
+    let chainOfTrust = await getArtifact(result, 'public/chain-of-trust.json');
+    let queue = new taskcluster.Queue();
+    let url = queue.buildUrl(queue.getArtifact, result.taskId, result.runId, 'public/chain-of-trust.json.sig');
+    let chainOfTrustSig = (await got(url, {encoding: null})).body;
+
+    let verifyKey = Buffer.from(fs.readFileSync('test/fixtures/ed25519_public_key', 'ascii'), 'base64');
+    assert(tweetnacl.sign.detached.verify(new Uint8Array(Buffer.from(chainOfTrust)), Buffer.from(chainOfTrustSig), verifyKey), 'ed25519 chain of trust signature does not appear to be valid');
```

Comment:
iiuc the `tweetnacl` docs correctly, the `Uint8Array` isn't necessary, since for nodejs >= v4 we can use the `Buffer` types.